src/main/java/com/agilethought/atsceapi/validator/user/UserValidator.java

```diff
@@ -94,6 +104,29 @@ private void validateEmail(String email) {
 
     }
 
+    private void validateUniqueEmailForNewUser(String email) {
+
+        if (userRepository.existsByEmail(email))
+            throw new BadRequestException(
+                    String.format(ALREADY_EXISTING_EMAIL, email)
+            );
+
+    }
+
+    private void validateUniqueEmailForExistingUser(String email, String id) {
+
+        Optional<User> userToBeUpdated = userRepository.findById(id);
```

Comment:
what happen if there are no users with given id?

Reply:
Before executing the methods from the UserValidator class, the
`public UpdateUserResponse updateUser(UpdateUserRequest request, String id) {`
method verifies that the provided id matches a user. When this doesn't happen, the API will return a 404 response with the message "User was not found with the given id:[provided_id]". Considering this, I think we can safely get the user being updated in this way to know if the existing email belongs to it or to another user.